code/modules/mob/living/carbon/human/species_types/android.dm

```diff
@@ -2,7 +2,7 @@
 	name = "Android"
 	id = SPECIES_ANDROID
 	say_mod = "states"
-	species_traits = list(NOBLOOD)
+	species_traits = list(NOBLOOD, TRAIT_CAN_USE_FLIGHTPOTION)
```

Comment:
Make this an `inherent_trait` instead, and change your if condition to just `HAS_TRAIT(exposed_carbon, TRAIT_CAN_USE_FLIGHTPOTION)`